### ISSUE TITLE
Eventing for Drawer component

### ISSUE THREAD

**Issue by antelopeb** [pearson-ux/web-components#264]
Per Pamela Mayne, we need to do some eventing on the drawer component. See list below for details.

Drawer
back button --- when the Back button is clicked
menu open click -- whenever the menu is opened
menu close click -- whenever the menu is closed

CSH [contextualized help specific]
External sites -- to any link outside the help/product
mail -- to any mailto link
pdf -- to any file with pdf extension
any embedded activation links/buttons
Topic links -- when a link to any help htm and html file is clicked

**Comment by davodey** [comment]
@antelopeb I dont think we need any custom events.  Devs can do standard event delegation and target the component like this:
```
standardDrawer.shadowRoot.addEventListener('click', event => {
  if (event.target.getAttribute('data-action') === 'close') {
    console.log('do something')
  }
})
```
Should be able to use this technique for all the list items pamela needs events for.
 
@dengeist Am I'm missing something?

**Comment by dengeist** [comment]
@antelopeb, @davodey: 

From my perspective, it seems like an antipattern to ask a dev to access the `.shadowRoot` from light DOM. I can add events for `open`, `back` and `close`. See a need for anything else?

Greg, would you mind explaining more of what you mean about the CSH stuff? I'm not following.

**Comment by antelopeb** [comment]
@dengeist -- CSH specific would be links that occur within the help content itself.

I think the general thought here is that we would be able capture these stats anytime someone implements, without having to ask them to implement that instrumentation themselves. That is, these would be non-configurable event capturing for our own tracking.
